### PR TITLE
better twitter validation

### PR DESCRIPTION
**Pull request by shrimalmadhur**
Fixes # .

### Motivation
- Add more strict `twitter` and `x` validations

### CODE REVIEW

**Review by joan-el — DISMISSED** [review]
this looks good, wait for Anto for final blessing

File: types/operator_metadata_test.go [hunk 1]
```diff
@@ -22,6 +22,16 @@ func TestOperatorMetadata(t *testing.T) {
 				Website:     "https://test.com",
 			},
 		},
+		{
```
Comment (joan-el) [inline types/operator_metadata_test.go:25]: love it you added this. as well

File: types/operator_metadata.go [hunk 2]
```diff
@@ -85,7 +85,25 @@ func (om *OperatorMetadata) Validate() error {
...
-func checkIfUrlIsValid(rawUrl string) error {
+func checkIfValidTwitterURL(twitterURL string) error {
+	// Basic validation
+	err := checkBasicURLValidation(twitterURL)
+	if err != nil {
+		return err
+	}
+
+	// Regular expression to validate URLs
+	urlPattern := regexp.MustCompile(`^(?:https?://)?(?:www\.)?twitter\.com/\w+|x\.com/\w+$`)
```
Comment (antojoseph) [inline types/operator_metadata.go:96]: Does this make sense?
`^(?:https?://)?(?:www\.)?(?:twitter\.com/\w+|x\.com/\w+)(?:/?|$)
`

**Review by antojoseph — COMMENTED** [review]
added a regex suggestion, please check

File: types/operator_metadata.go (hunk 2 above)
Comment (shrimalmadhur) [reply types/operator_metadata.go:96]: why would twitter have `:` in it @antojoseph ? can you give an example of what url you are thinking?

Comment (shrimalmadhur) [reply types/operator_metadata.go:96]: oh I se. lol I think i read it wrong

Comment (antojoseph) [reply types/operator_metadata.go:96]: we can remove that, i don't think ppl specify their twitter links like https://twitter.com:443/abcd

Comment (shrimalmadhur) [reply types/operator_metadata.go:96]: It makes sense

Comment (shrimalmadhur) [reply types/operator_metadata.go]: updated

**Review by antojoseph — APPROVED** [review]
lgtm

**Pull request merged**